docs/source/running-jobs/many-jobs.rst

```diff
@@ -0,0 +1,273 @@
+      --do_train \
+      --max_seq_length 128 \
+      --per_device_train_batch_size 32 \
+      --learning_rate ${LR} \
+      --max_steps ${MAX_STEPS} \
+      --output_dir /tmp/imdb/ --overwrite_output_dir \
+      --fp16
+
+.. raw:: html
+    
+    </details>
+
+We can now launch a job with different hyperparameters by specifying the envs.
+
+.. code-block:: bash
+
+  sky launch -c dnn dnn-template.yaml \
+    --env LR=1e-5 \
+    --env MAX_STEPS=100
+
+Or, you can store the envs in a dotenv file and launch the job with the file: ``configs/job1.env``.
+
+.. code-block:: bash
+
+  # configs/job1.env
+  LR=1e-5
+  MAX_STEPS=100
+
+.. code-block:: bash
+
+  sky launch -c dnn dnn-template.yaml \
+    --env-file configs/job1.env
+
+
+
+2. Track Job Output
+~~~~~~~~~~~~~~~~~~~
+
+When running many jobs, it is useful to track live outputs of each job. We recommend using WandB to track the outputs of all jobs.
+
+.. raw:: html
+
+    <details>
+    <summary>SkyPilot YAML with WandB: <code>dnn-template.yaml</code></summary>
+
+.. code-block:: yaml
+  :emphasize-lines: 7-7,19-19,34-34
+
+  # dnn-template.yaml
+  name: huggingface
+
+  envs:
+    LR: 2e-5
+    MAX_STEPS: 50
+    WANDB_API_KEY: # Empty field means this field is required when launching the job.
+      
+  resources:
+    accelerators: V100:4
+
+  setup: |
+    set -e  # Exit if any command failed.
+    git clone https://github.com/huggingface/transformers/ || true
+    cd transformers
+    pip install .
+    cd examples/pytorch/text-classification
+    pip install -r requirements.txt torch==1.12.1+cu113 --extra-index-url https://download.pytorch.org/whl/cu113
+    pip install wandb
+
+  run: |
+    set -e  # Exit if any command failed.
+    cd transformers/examples/pytorch/text-classification
+    python run_glue.py \
+      --model_name_or_path bert-base-cased \
+      --dataset_name imdb  \
+      --do_train \
+      --max_seq_length 128 \
+      --per_device_train_batch_size 32 \
+      --learning_rate ${LR} \
+      --max_steps ${MAX_STEPS} \
+      --output_dir /tmp/imdb/ --overwrite_output_dir \
+      --fp16 \
+      --report_to wandb
+
+.. raw:: html
+
+    </details>
+
+We can now launch the job with the following command (``WANDB_API_KEY`` should existing in your local environment variables).
+
+.. code-block:: bash
+
+  sky launch -c dnn dnn-template.yaml \
+    --env-file configs/job1.env \
+    --env WANDB_API_KEY
+
+
+
+Scale up the Job
+-----------------
+
+With the above setup, we can now scale up a job to many in-parallel jobs by creating multiple config files and
+submitting them with :ref:`SkyPilot managed jobs <managed-jobs>`.
+
+We create a config file for each job in the ``configs`` directory.
+
+.. code-block:: bash
+
+  # configs/job1.env
+  LR=1e-5
+  MAX_STEPS=100
+
+  # configs/job2.env
+  LR=2e-5
+  MAX_STEPS=200
+
+  ...
+
+We can then submit all jobs by iterating over the config files.
+
+.. code-block:: bash
+
+  for config_file in configs/*.env; do
```

Comment:
One question in mind was "how to write out a config file for each param combo". Is it better to show a representative python program to iterate over params, then either (1) write out the env files, or (2) directly call jobs launch Python API / CLI from within Python script?

Reply:
Added a python script for generating the configs.